src/gui/editor/Editor.ts

```diff
@@ -14,314 +14,317 @@ type Style = "b" | "i" | "u" | "c" | "a"
-			if (olIndex > -1) {
-				this.styles.listing = "ol"
-			} else {
-				this.styles.listing = null
-			}
-		} else if (olIndex === -1) {
-			if (ulIndex > -1) {
-				this.styles.listing = "ul"
-			} else {
-				this.styles.listing = null
-			}
-		} else if (olIndex > ulIndex) {
-			this.styles.listing = "ol"
-		} else {
-			this.styles.listing = "ul"
-		}
-
-		//links
-		this.styles.a = pathSegments.includes("A")
-		// alignment
-		let alignment = pathSegments.find(f => f.includes("align"))
-
-		if (alignment !== undefined) {
-			switch (alignment.split(".")[1].substring(6)) {
-				case "left":
-					this.styles.alignment = "left"
-					break
-
-				case "right":
-					this.styles.alignment = "right"
-					break
-
-				case "center":
-					this.styles.alignment = "center"
-					break
-
-				default:
-					this.styles.alignment = "justify"
-			}
-		} else {
-			this.styles.alignment = "left"
-		}
-
-		// font
-		this.styles.c = pathSegments.find(f => f.includes("monospace")) !== undefined
-		// decorations
-		this.styles.b = this._squire.hasFormat("b")
-		this.styles.u = this._squire.hasFormat("u")
-		this.styles.i = this._squire.hasFormat("i")
-	}
-
-	makeLink() {
-		Dialog.showTextInputDialog("makeLink_action", "url_label", null, "").then(url => {
-			if (isMailAddress(url, false)) {
-				url = "mailto:" + url
-			} else if (!url.startsWith("http://") && !url.startsWith("https://") && !url.startsWith("mailto:") && !url.startsWith("{")) {
-				url = "https://" + url
-			}
-
-			this._squire.makeLink(url)
-		})
-	}
-
-	insertImage(srcAttr: string, attrs?: Record<string, string>): HTMLElement {
-		return this._squire.insertImage(srcAttr, attrs)
-	}
-
-	/**
-	 * Inserts the given html content at the current cursor position.
-	 */
-	insertHTML(html: string) {
-		this._squire.insertHTML(html)
-	}
-
-	getDOM(): HTMLElement {
-		return this._squire.getRoot()
-	}
-
-	getCursorPosition(): ClientRect {
-		return this._squire.getCursorPosition()
-	}
-
-	focus(): void {
-		this._squire.focus()
-
-		this.getStylesAtPath()
-	}
-
-	isAttached(): boolean {
-		return this._squire != null
-	}
-
-	removeAllFormatting(): void {
-		// Create a range which contains the whole editor
-		const range = document.createRange()
-		range.selectNode(this._squire.getRoot())
-
-		this._squire.removeAllFormatting(range)
-	}
-
-	getSelectedText(): string {
-		return this._squire.getSelectedText()
-	}
-
-	addEventListener(type: string, handler: (arg0: Event) => void) {
-		this._squire.addEventListener(type, handler)
-	}
-
-	setSelection(range: Range) {
-		this._squire.setSelection(range)
-	}
+    _squire: Squire
+    view: Component["view"]
+    onbeforeupdate: Component["onbeforeupdate"]
+    onremove: Component["onremove"]
+    initialized: DeferredObject<void>
+    _domElement: HTMLElement | null = null
+    _enabled: boolean
+    private _active: boolean
+    private _createsLists: boolean
+    private _minHeight: number | null
+    private _flexHeight: boolean
```

Comment:
This doesn't need to be a field since it's only ever accessed from the constructor (this might be a redundant point anyway, see other comment)